ts/WoltLabSuite/Core/Ui/Color/Picker.ts

```diff
@@ -67,14 +42,217 @@ class UiColorPicker {
       },
       options || {},
-    );
+    ) as ColorPickerOptions;
+
+    element.addEventListener("click", () => this.openPicker());
+  }
+
+  public _dialogSetup(): ReturnType<DialogCallbackSetup> {
+    return {
+      id: `${DomUtil.identify(this.element)}_colorPickerDialog`,
+      source: `
+<div class="colorPickerDialog">
+  <div class="row rowColGap formGrid">
+    <div class="col-xs-12 col-md-8">
+      <dl>
+        <dt>${Language.get("wcf.style.colorPicker.color")}</dt>
+        <dd>
+          <input type="color">
+        </dd>
+      </dl>
+      <dl>
+        <dt>${Language.get("wcf.style.colorPicker.alpha")}</dt>
+        <dd>
+          <input type="range" min="0" max="1" step="0.01">
+        </dd>
+      </dl>
+      <dl>
+        <dt>${Language.get("wcf.style.colorPicker.hexAlpha")}</dt>
+        <dd>
+          <div class="inputAddon">
+            <span class="inputPrefix">#</span>
+            <input type="text" class="medium">
+          </div>
+        </dd>
+      </dl>
+    </div>
+    <div class="col-xs-12 col-md-4 colorPickerComparison">
+      <small>${Language.get("wcf.style.colorPicker.new")}</small>
+      <div class="colorPickerColorNew">
+        <span style="background-color: ${this.input.value}"></span>
+      </div>
+      <div class="colorPickerColorOld">
+        <span style="background-color: ${this.input.value}"></span>
+      </div>
+      <small>${Language.get("wcf.style.colorPicker.current")}</small>
+    </div>
+  </div>
+  <div class="formSubmit">
+    <button class="buttonPrimary">${Language.get("wcf.style.colorPicker.button.apply")}</button>
+  </div>
+</div>`,
+      options: {
+        onSetup: (content) => {
+          this.colorInput = content.querySelector("input[type=color]") as HTMLInputElement;
+          this.colorInput.addEventListener("input", () => this.updateColor());
+          this.alphaInput = content.querySelector("input[type=range]") as HTMLInputElement;
+          this.alphaInput.addEventListener("input", () => this.updateColor());
+
+          this.newColor = content.querySelector(".colorPickerColorNew > span") as HTMLSpanElement;
+          this.oldColor = content.querySelector(".colorPickerColorOld > span") as HTMLSpanElement;
+
+          this.colorTextInput = content.querySelector("input[type=text]") as HTMLInputElement;
+          this.colorTextInput.addEventListener("blur", (ev) => this.updateColorFromHex(ev));
+          this.colorTextInput.addEventListener("keypress", (ev) => this.updateColorFromHex(ev));
+
+          content.querySelector(".formSubmit > .buttonPrimary")!.addEventListener("click", () => this.submitDialog());
+
+          if (ColorUtil.isValidColor(this.input.value)) {
+            this.setInitialColor(this.input.value);
+          } else if (this.element.dataset.color && ColorUtil.isValidColor(this.element.dataset.color)) {
+            this.setInitialColor(this.element.dataset.color);
+          } else {
+            this.setInitialColor("#FFF0");
```

Comment:
Should alpha be 0 or F?

Reply:
No, `0` is correct because by default no color is set.